contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1067,7 +995,10 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
 
             // INTENTIONAL_REVERT needs to pass up the user-provided return data encoded into the
             // flag, *not* the full encoded flag. All other revert types return no data.
-            if (flag == RevertFlag.INTENTIONAL_REVERT) {
+            if (
+                flag == RevertFlag.INTENTIONAL_REVERT
+                || _isCreate
```

Comment:
For consistency, the `_handleExternalCreate` function will return an `_encodeRevertData` as if the subcall reverted, even in the case of `CREATE_COLLISION` or `UNSAFE_BYTECODE`.